doc/api/node_report.md

```diff
@@ -0,0 +1,211 @@
+error object must be the second parameter.
+
+```js
+try {
+  process.chdir('/non-existent-path');
+} catch (err) {
+  report.triggerReport(filename, err);
+}
+// any other code
+```
+
+The content of the diagnostic report can be returned as a JSON-formatted Object
+via an API call from a JavaScript application:
+
+```js
+const { report } = require('util');
+const str = report.getReport();
+console.log(str);
+```
+
+The API takes an optional additional argument `err` - an Error object that will
+be used as the context for the JavaScript stack printed in the report.
+
+```js
+const { report } = require('util');
+const str = report.getReport(new Error('custom error'));
+console.log(str);
+```
+
+The API versions are useful when inspecting the Runtime state from within
+the application, in expectation of self-adjusting the resource consumption,
+load balancing, monitoring etc.
+
+Content of the report consists of a header section containing the event
+type, date, time, PID and Node.js version, sections containing JavaScript and
+native stack traces, a section containing V8 heap information, a section
+containing `libuv` handle information and an OS platform information section
+showing CPU and memory usage and system limits. An example report can be
+triggered using the Node.js REPL:
+
+```raw
+$ node
+> const { report } = require('util');
+> report.triggerReport();
+Writing Node.js report to file: node-report.20181126.091102.8480.001.json
+Node.js report completed
+>
+```
+
+When a report is triggered, start and end messages are issued to stderr
+and the filename of the report is returned to the caller. The default filename
+includes the date, time, PID and a sequence number. The sequence number helps
+in associating the report dump with the runtime state if generated multiple
+times against the same Node.js process.
+
+## Configuration
+
+Additional runtime configuration that influences the report generation
+constraints are available using `setDiagnosticReportOptions` API.
+
+```js
+const { report } = require('util');
+report.setDiagnosticReportOptions({
+  events: ['exception', 'fatalerror', 'signal'],
+  signal: 'SIGUSR2',
+  filename: 'stderr',
+  directory: '/home/nodeuser',
+  verbose: true
+});
+```
+
+`events` array contain one or more of report triggering options.
+Only valid values are `exception`, `fatalerror` and `signal`.
+By default, a report is not produced on any of these events.
+
+`signal` specifies the POSIX signal identifier that will be used
+to intercept external triggers for report generation. Defaults to
+`SIGUSR2`.
+
+`filename` specifies the name of the output file in the file system.
+special meaning is attached to `stdout` and `stderr`. Usage of these
+will result in report being written to the associated standard streams.
+No translations occur to the supplied string, so URL protocols are not
+supported. Defaults to a composite filename that contain timestamp, pid and
```

Comment:
I’d remove the `No translations …` sentence.
Also:

```suggestion
supported. Defaults to a composite filename that contains the timestamp, pid and
```

Reply:
The background to the `No translations occur..` statement is is response to @jasnell 's proposal (https://github.com/nodejs/node/pull/22712#discussion_r215334799) to cover that case specifically. Do you think it is confusing, or not required at all?

Reply:
I think “URLs are not supported” would be sufficient here, because it’s not generally clear what a translation is in this context?

Reply:
made this change.